pkg/vendir/directory/staging_dir.go

```diff
@@ -58,7 +61,74 @@ func (d StagingDir) NewChild(path string) (string, error) {
 	return childPath, nil
 }
 
+func (d StagingDir) CopyExistingFiles(rootDir string, stagingPath string, contents ctlconf.DirectoryContents) error {
+
+	if len(contents.IgnorePaths) == 0 {
+		return nil
+	}
+
+	// Create reference point from staging path to root
+	rootPath := strings.Replace(stagingPath, d.stagingDir, rootDir, 1)
```

Comment:
im still chewing on this line. will play around more.

Reply:
Great, keep me posted on what you would suggest. I messed around with this a bit and was the best solution I could come up with. Not super happy with the solution.